### PR TITLE
Add a pass that drops export ops marked as excluded during the VMToEmitC conversion

### PR DESCRIPTION
**Pull request by simon-camp**

### CODE REVIEW

File: iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp [hunk 1]
```diff
@@ -699,12 +700,20 @@ static LogicalResult canonicalizeModule(
     modulePasses.addPass(mlir::createCanonicalizerPass());
   }
 
+  // C target specific pass
+
+  // Erase exports annotated with 'emitc.exclude'. This makes testing
+  // of partially supported ops easier. For the DCE pass to remove the
+  // referenced function it must be unused and marked private.
```
Comment (marbre) [inline iree/compiler/Dialect/VM/Target/C/CModuleTarget.cpp:707]: NIT: Maybe the last sentence could be reformulated? Don't have a suggestion though.

**Review by marbre — APPROVED** [review]
Awesome! This makes a look easier to add further conversions. I tested locally and works as is should.

File: iree/compiler/Dialect/VM/Target/C/test/constant_ops.mlir [hunk 2]
```diff
@@ -14,8 +14,8 @@ vm.module @constant_ops {
   // CHECK-NEXT: iree_vm_buffer_t rodata_buffers[2];
   // CHECK-NEXT: };
 
-  vm.rodata private @buffer_1 dense<[1, 2, 3]> : tensor<3xi8>
-  vm.rodata private @buffer_2 dense<[1, 2, 3]> : tensor<3xi32>
+  vm.rodata public @buffer_1 dense<[1, 2, 3]> : tensor<3xi8>
```
Comment (benvanik) [inline iree/compiler/Dialect/VM/Target/C/test/constant_ops.mlir:17]: rodata should in general be private - there's no way to reference them outside of the module (can't export rodata) and it's better to have symbol DCE be able to drop them

File: iree/vm/test/assignment_ops_f32.mlir [hunk 3]
```diff
@@ -1,21 +1,21 @@
-vm.module @assignment_ops_i64 {
+vm.module @assignment_ops_f32 {
 
   //===--------------------------------------------------------------------===//
   // ExtF32: Conditional assignment
   //===--------------------------------------------------------------------===//
 
-  vm.export @test_select_f32
-  vm.func @test_select_f32() {
+  vm.export @test_select_f32  attributes {emitc.exclude}
```
Comment (benvanik) [inline iree/vm/test/assignment_ops_f32.mlir:7]: nit: extra space

**Review by benvanik — APPROVED** [review]

File: iree/compiler/Dialect/VM/Target/C/test/constant_ops.mlir (hunk 2 above)
Comment (simon-camp) [reply iree/compiler/Dialect/VM/Target/C/test/constant_ops.mlir:20]: Sure but I don't want DCE to kick in here. I shamelessly copied your formulation to a comment :P

**Review by benvanik — APPROVED** [review]

**Pull request merged**